File: python/program/luxury.py
from mobil import Mobil

class Luxury(Mobil):
    def __init__(self, nama, merk, mesin, seat, warna, tahun, harga, fiturTeknologi, fiturKenyamanan, materialInterior):
        super().__init__(nama, merk, mesin, seat, warna, tahun, harga)
        self.fiturTeknologi = fiturTeknologi
        self.fiturKenyamanan = fiturKenyamanan
        self.materialInterior = materialInterior

    def getFiturTeknologi(self):
        return self.fiturTeknologi

    def getFiturKenyamanan(self):
        return self.fiturKenyamanan

    def getMaterialInterior(self):
        return self.materialInterior

    def getFiturTeknologi(self):
        return self.fiturTeknologi

    def setFiturTeknologi(self, fiturTeknologi):
        self.fiturTeknologi = fiturTeknologi

    def setFiturKenyamanan(self, fiturKenyamanan):
        self.fiturKenyamanan = fiturKenyamanan

    def setMaterialInterior(self, materialInterior):
        self.materialInterior = materialInterior

    def tampilkan_info(self):
        """Method override untuk menampilkan informasi mobil luxury"""
        super().tampilkan_info()
        print(f"Fitur Teknologi : {self.fiturTeknologi}")
        print(f"Fitur Kenyamanan: {self.fiturKenyamanan}")
        print(f"Material Interior: {self.materialInterior}")

File: java/program/Luxury.java
class Luxury extends Mobil{
    private String fiturTeknologi;
    private String fiturKenyamanan;
    private String materialInterior;

    public Luxury() {
        super();
        this.fiturTeknologi = "";
        this.fiturKenyamanan = "";
        this.materialInterior = "";
    }

    public Luxury(String nama, String merk, Mesin mesin, int seat, String warna, int tahun, String harga, String fiturTeknologi, String fiturKenyamanan, String materialInterior) {
        super(nama, merk, mesin, seat, warna, tahun, harga);
        this.fiturTeknologi = fiturTeknologi;
        this.fiturKenyamanan = fiturKenyamanan;
        this.materialInterior = materialInterior;
    }

    public String getFiturTeknologi() {
        return fiturTeknologi;
    }

    public String getFiturKenyamanan() {
        return fiturKenyamanan;
    }

    public String getMaterialInterior() {
        return materialInterior;
    }

    public void setFiturTeknologi(String fiturTeknologi) {
        this.fiturTeknologi = fiturTeknologi;
    }

    public void setFiturKenyamanan(String fiturKenyamanan) {
        this.fiturKenyamanan = fiturKenyamanan;
    }

    public void setMaterialInterior(String materialInterior) {
        this.materialInterior = materialInterior;
    }

    @Override
    public void tampilkanInfo() {
        super.tampilkanInfo();
        System.out.println("Fitur Teknologi: " + fiturTeknologi);
        System.out.println("Fitur Kenyamanan: " + fiturKenyamanan);
        System.out.println("Material Interior: " + materialInterior);
    }
}